build-logic/src/main/groovy/qsl.library.gradle

```diff
@@ -0,0 +1,135 @@
+/*
+ * Buildscript for defining a QSL library.
+ *
+ * This is a convention plugin, and is applied by adding the following to your `plugins` block: `id("qsl.library")`.
+ * A library is not exactly a mod, but rather a maven artifact that depends on a set of modules. For example, the `core`
+ * library may depend on a module to provide the event code, code to add Quilt mods to the crash report, etc.
+ *
+ * This buildscript declares an extension available under the `qslLibrary` name. This extension is used to configure
+ * attributes of the library, namely the name of the library and the library's version. The buildscript also declares
+ * some additional tasks that can be used to run the game and runTestmodClient, runTestmodServer. These testmod tasks
+ * can be run using `gradle LIBRARY_NAME:TASK_NAME`.
+ */
+
+import qsl.internal.Git
+import qsl.internal.extension.QSLLibraryExtension
+
+plugins {
+	id("maven-publish")
+	id("net.kyori.indra.git")
+	id("qsl.common")
+}
+
+def extension = project.extensions.create("qslLibrary", QSLLibraryExtension, project)
+
+// We seem to have to use afterEvaluate to verify this stuff
+afterEvaluate {
+	// TODO: Validate the extension parts
+}
```

Comment:
figure out what's going on with this

Reply:
I believe I have an idea why:

This is simply based on plugin load order:

Plugin is loaded and empty extension is instantiated.

Entire convention buildscript is run.

extension is loaded.

Because of the above steps, this is required

Reply:
@zml2008 could you explain if I am correct above?

Reply:
nvm, misunderstood the comment, you're fine